File: backoffice/src/product/domain/exceptions/InvalidProductIdException.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain import DomainException

"""
 *
 * Class 
 *
"""

class InvalidProductIdException( DomainException ):

    """
     *
     * Constants 
     *
    """

    __INVALID_PRODUCT_ID = 109

    """
     *
     * Methods 
     *
    """

    def __init__( self, id : str ) -> None:
        super().__init__( 
            self.__INVALID_PRODUCT_ID,
            'The product id {} is invalid'.format( id )
        )

File: preparation/src/product/domain/exceptions/InvalidProductIdException.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.product.domain {

    public class InvalidProductIdException : DomainException {

        /* 
         * 
         * Constants
         *
        */

        private const int INVALID_PRODUCT_ID = 109;

        /* 
         * 
         * Methods
         *
        */

        public InvalidProductIdException( string id ) : base( 
            INVALID_PRODUCT_ID,
            string.Format( "The product id {0} is invalid", id )
        ) {
        }

    }

}